### DISCUSSION TITLE
Linux bundle build works but when run returns license not found

### DISCUSSION THREAD

**Discussion by rnmixon** [airsdk/Adobe-Runtime-Support#2458] [General]
I was able to build a linux bundle with a simple application with help from this post [https://github.com/airsdk/Adobe-Runtime-Support/discussions/2419#discussioncomment-4674862](https://github.com/airsdk/Adobe-Runtime-Support/discussions/2419#discussioncomment-4674862).

Now I am trying to build a bundle for a more complicated application - but this time using a ".airi" file as input. It works fine on the PC/Windows environment and on the Linux environment the build appears to go successfully. However when I run the application I get this error:

```
*******************************************************************************
Adobe AIR - Linux runtime is limited to use by commercially licensed developers
*******************************************************************************
```
This is different than the "Failed to load the AIR Runtime" error I got earlier that was resolved by placing my adt.lic file under ~/.airsdk folder.

Here is the command I use to do the build/prepare:
```
    ${adt} -package \
		-keystore ${key_store} -storetype pkcs12 -storepass ${passphr} -tsa none \
		-target bundle dist/linux "${PROG_NAME_BASE}.airi"
```
Also there was a similar post from a colleague almost a year ago, but they  were using ".swf" and "*-app.xml" files as input to the adt prepare, I need to use an ".airi" file that is created on the PC and then moved to the linux system for final preparation. [https://github.com/airsdk/Adobe-Runtime-Support/discussions/1852#discussion-4017663](https://github.com/airsdk/Adobe-Runtime-Support/discussions/1852#discussion-4017663)

Any ideas on how to resolve this?

Thank you much - Richard

**Comment by rnmixon** [comment]
I also checked that I have the adt.lic file in both suggested locations and other settings appear correct:

```
n2kops@dev2n2k:~/n2kSrc/Maretron N2KView BlackBox/dist/linux$ echo $AIR_HOME
/home/n2kops/tools/apache-flex-sdk-4.16.1-bin
n2kops@dev2n2k:~/n2kSrc/Maretron N2KView BlackBox/dist/linux$ echo $PATH
/home/n2kops/tools/apm_1.2.0:/home/n2kops/tools/apache-flex-sdk-4.16.1-bin/bin:/home/n2kops/.local/bin:/home/n2kops/tools/apm_1.2.0:/home/n2kops/tools/apache-flex-sdk-4.16.1-bin/bin:/home/n2kops/.local/bin:/home/n2kops/tools/apm_1.2.0:/home/n2kops/tools/apache-flex-sdk-4.16.1-bin/bin:/home/n2kops/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
n2kops@dev2n2k:~/n2kSrc/Maretron N2KView BlackBox/dist/linux$ adt -license
AIR Developer Tool version 50.1.1.2
License file is present and valid
DeveloperID=fbb267e3-fa88-41f3-be56-b4208d213ddd

n2kops@dev2n2k:~/n2kSrc/Maretron N2KView BlackBox/dist/linux$ ls -l ~/.airsdk/adt.lic ${AIR_HOME}/lib/adt.lic
-rw-r--r-- 1 n2kops n2kops 142 Feb  7 17:53 /home/n2kops/.airsdk/adt.lic
-rw-r--r-- 1 n2kops n2kops 142 Feb  8 16:00 /home/n2kops/tools/apache-flex-sdk-4.16.1-bin/lib/adt.lic
n2kops@dev2n2k:~/n2kSrc/Maretron N2KView BlackBox/dist/linux$
```

Any ideas on what else I can check?

Thank you - Richard

**Reply by ajwfrost** [reply]
Did you package up the .airi file also on the same computer, or on a different machine? Are you able to check the contents using:
```
$ unzip -l "${PROG_NAME_BASE}.airi"
```
and make sure there's a `license.txt` file listed there?

If you're still having problems, can you email us (adobe.support) and send the .airi file, we can try this here? Currently we're not able to get this to reproduce..

thanks

**Reply by rnmixon** [reply]
That was it! Thank you. The ".airi" file was package on a different (Windows) computer - and the resulting ".airi" file does NOT have the license.

I tried adding the license.txt file to the .airi file using zip utilities - seems to have worked. When I move the .airi file to Linux and build the bundle I can see the license.txt file under META-INF/AIR - but I still get the error when I run it:

`Adobe AIR - Linux runtime is limited to use by commercially licensed developers`

I'll try re-building the .airi file instead of patching it. 
Question: On Windows where do you place the adt.lic file: in %AIR_HOME%\lib or in %HOMEPATH%\.airsdk

Thank you much - Richard

**Reply by ajwfrost** [reply]
Yes patching probably won't work :-) 
If you're on a reasonably recent version of AIR SDK then `%HOMEPATH%\.airsdk\adt.lic` should work

thanks

**Comment by rnmixon** [comment]
My colleague was not able to get Flash Builder to use the license file in %HOMEPATH%\.airsdk\adt.lic .

Their time is scarce so I am trying to build the .swf, .airi and bundle in a script. I am having issues but will open a separate post to ask about them.

Thank you again - Richard

**Reply by ajwfrost** [reply]
Which version of the AIR SDK are they using? If it's fairly old then the license should be in the AIRSDK\lib folder..

If you email me the .airi file and let me know your developer ID, we can try hacking it a little here....

**Reply by rnmixon** [reply]
Andrew,
Thank you - but getting permission to send the ".airi" file is complicated and I'm not able to do it in a timely manner.

So they've asked me to just try and get this working as a scripted build. I started a new post on issues I'm having with that process so as to keep the issues separated. Here the link is: [https://github.com/airsdk/Adobe-Runtime-Support/discussions/2470#discussion-4867312](https://github.com/airsdk/Adobe-Runtime-Support/discussions/2470#discussion-4867312)

Thank you much as always - Richard

**Reply by rnmixon** [reply]
BTW, I think I did find out the version of the Harman AIR SDK being used. Does 32.0 make sense. They last updated on May 26, 2021. 

In the root I found the airsdk.xml file with the following tag:
`<airSdk xmlns="http://ns.adobe.com/air/sdk/32.0">`

Would that version have looked for adt.lic in folder ~/.airsdk  or AIRSDK\lib ?

Thank you - Richard

**Reply by ajwfrost** [reply]
Ah, so 32.0 isn't one that we did, that's the Adobe version. So that wouldn't have any license file or license generation in it....
You can use the AIR SDK 32 to turn that into a bundle for distribution on Windows and macOS, it's just that they didn't support Linux!

If this is still causing a problem then I'm wondering whether we could update the AIRI -> AIR process to look for this scenario and inject an appropriate license file if the machine on which the conversion happens is one with our licensed SDK on it... so then, it wouldn't matter that the original AIRI build was done with Adobe's SDK, the result would be packaged with our runtime files and licensed on that basis...
